aten/src/ATen/native/TensorShape.cpp

```diff
@@ -706,6 +706,11 @@ Tensor slice(const Tensor& self, int64_t dim, int64_t start, int64_t end, int64_
   auto strides = self.strides().vec();
   // TODO: support negative strides
   TORCH_CHECK(step > 0, "slice step must be positive");
+
+  // INT64_MAX stands for default value.
+  if (start == INT64_MAX) {
```

Comment:
Why don't we have the same adjustment of start according to step as we have in lists ?
```
  if (*start == INT64_MAX) {
    *start = (step < 0) ? INT64_MAX : 0;
  }
```

Also don't we need to do the same adjustment of stop ?
```
  if (*stop == INT64_MAX) {
    *stop = (step < 0) ? INT64_MIN : INT64_MAX;
  }
```

Reply:
Stop being maxint is a preexisting condition. Since we only support step > 0 here, stop should be INT64_MAX if not specified.

I've asked Edward if he'd like me to undertake supporting step < 0 here, but seems like the project is more complicated than just changing impl like it was done for lists. Negative tensor striding has to be supported on native level - we can't simply copy values here.